app/main.py

```diff
@@ -1,2 +1,86 @@
 class Dictionary:
+
+    def __init__(self):
+        self.capacity = 8
+        self.threshold = int(self.capacity * (2 / 3))
+        self.length = 0
+        self.base_data = [None for i in range(self.capacity)]
+
+    def __setitem__(self, key, value):
+
+        if self.length >= self.threshold:
+            self.resize()
+
+        index_ = int(hash(key) % self.capacity)
+
+        while self.base_data[index_] is not None:
+
+            if self.base_data[index_][0] == key and\
```

Comment:
In `if` you should compare hashes first because it's faster than comparing keys (e.x. they could be long tuples).